crates/transaction-pool/benches/reorder.rs

```diff
@@ -0,0 +1,198 @@
+                &mut group,
+                "VecTxPoolSortStable",
+                txs.clone(),
+                new_txs.clone(),
+                base_fee,
+            );
+
+            // Unstable sorting of unsorted collection
+            txpool_reordering_bench::<VecTxPoolSortUnstable>(
+                &mut group,
+                "VecTxPoolSortUnstable",
+                txs.clone(),
+                new_txs.clone(),
+                base_fee,
+            );
+
+            // Vanilla sorting with sorted insert
+            txpool_reordering_bench::<VecTxPoolSortStableAndOnInsert>(
+                &mut group,
+                "VecTxPoolSortStableAndOnInsert",
+                txs,
+                new_txs,
+                base_fee,
+            );
+        }
+    }
+}
+
+fn txpool_reordering_bench<T: BenchTxPool>(
+    group: &mut BenchmarkGroup<WallTime>,
+    description: &str,
+    seed: Vec<MockTransaction>,
+    new_txs: Vec<MockTransaction>,
+    base_fee: u64,
+) {
+    let setup = || {
+        let mut txpool = T::default();
+        txpool.reorder(base_fee);
+
+        for tx in seed.iter() {
+            txpool.add_transaction(tx.clone());
+        }
+        (txpool, new_txs.clone())
+    };
+
+    let group_id = format!(
+        "txpool | seed size: {} | input size: {} | {}",
+        seed.len(),
+        new_txs.len(),
+        description
+    );
+    group.bench_function(group_id, |b| {
+        b.iter_with_setup(setup, |(mut txpool, new_txs)| {
+            black_box({
+                // Reorder with new base fee
+                let bigger_base_fee = base_fee.saturating_add(10);
+                txpool.reorder(bigger_base_fee);
+
+                // Reorder with new base fee after adding transactions.
+                for new_tx in new_txs {
+                    txpool.add_transaction(new_tx);
+                }
+                let smaller_base_fee = base_fee.saturating_sub(10);
+                txpool.reorder(smaller_base_fee);
+            })
+        });
+    });
+}
+
+fn generate_test_data(
+    seed_size: usize,
+    input_size: usize,
+) -> (Vec<MockTransaction>, Vec<MockTransaction>, u64) {
+    let config = ProptestConfig::default();
+    let mut runner = TestRunner::new(config);
+
+    let txs = prop::collection::vec(any::<MockTransaction>(), seed_size)
+        .new_tree(&mut runner)
+        .unwrap()
+        .current();
+
+    let new_txs = prop::collection::vec(any::<MockTransaction>(), input_size)
+        .new_tree(&mut runner)
+        .unwrap()
+        .current();
+
+    let base_fee = any::<u64>().new_tree(&mut runner).unwrap().current();
+
+    (txs, new_txs, base_fee)
+}
+
+mod implementations {
+    use reth_transaction_pool::PoolTransaction;
+
+    use super::*;
+
+    /// This implementation appends the transactions and uses [Vec::sort_by] function for sorting.
+    #[derive(Default)]
+    pub struct VecTxPoolSortStable {
+        inner: Vec<MockTransaction>,
```

Comment:
why are we benching vec? do we want to transition to binaryhead? if so we should use that 

what are the numbers for btreemap,